test/jdk/java/lang/module/ClassFileVersionsTest.java

```diff
@@ -99,7 +100,11 @@
                 { 60,   0,  Set.of(TRANSITIVE) },
                 { 60,   0,  Set.of(STATIC, TRANSITIVE) },
 
-                { 61,   0,  Set.of()},                       // JDK 17
+                { 61,   0,  Set.of(STATIC) },                // JDK 17
+                { 61,   0,  Set.of(TRANSITIVE) },
+                { 61,   0,  Set.of(STATIC, TRANSITIVE) },
+
+                { 62,   0,  Set.of()},                       // JDK 18
```

Comment:
This seems unduly repetitive. Could this be dynamically generated, perhaps in a future release?

Reply:
I've had similar thoughts; that strikes me as a fine RFE for after this fork. I see what the code is doing, but haven't delved into the module system details to understand exactly the rationale for these tests. In any case, filed the RFE JDK-8257856: "Make ClassFileVersionsTest.java robust to JDK version updates."

Reply:
There was a change to JVMS 4.7.25 in Java 10 to add a rule for the requires_flags that are allowed. This is why this test started was created to test 53.0 vs. 54.0 class files. It wasn't intended to be a burden to update at each release so I'll re-implement it.